### PR TITLE
Drop and re-create genesis_info_view

### LINKED ISSUE #685: [bug]: Can't transfer assets: unknown sqlite error: SQL logic error: no such column: anchor_txid (1)
### Background

Describe your issue here.

### Your environment

```sh
tapcli getinfo                       # version of `tapd`, `lnd`, and network
```

```
{
    "version": "0.3.1-alpha commit=v0.3.1-21-gb8a007d",
    "lnd_version": "0.16.2-beta",
    "network": "testnet3",
    "lnd_identity_pubkey": "0332c0638eddcc313465ad993b482d290619f5b029dc9d3f0bdd9f83f12b27ead5",
    "node_alias": "0332c0638eddcc313465",
    "block_height": 2539030,
    "block_hash": "0000000000000023bd649f64a75eb30d6fe4c4b794d947fea8eb16d5619c142f",
    "sync_to_chain": true
}
```

```
uname -mrsv                          # operating system 
```

```
Linux 6.1.27-43.48.amzn2023.x86_64 #1 SMP PREEMPT_DYNAMIC Tue May  2 04:53:36 UTC 2023 x86_64
```

```
bitcoind --version
```

```
Bitcoin Core version v22.0.0
```

### Steps to reproduce

1.) Mint assets on the old version of tapd

2.) Try sending assets on the new version of tapd
Resulting error:
```
[tapcli] unable to make addr: rpc error: code = Unknown desc = unable to make new addr: unable to make address for unknown asset 53b929739ef4db7554b484071999f59d7d74e935584fcc98005b581b427e0c9d: unknown sqlite error: SQL logic error: no such column: anchor_txid (1)
```
### Expected behavior

Successful transfer of assets

### Actual behavior

I get an error

### PR DESCRIPTION
**Pull request by GeorgeTsagk**
Closes #685

### CODE REVIEW

**Review by ffranr — COMMENTED** [review]
I think we should add a word or two in the commit body to explain why this change is necessary.

File: tapdb/sqlc/migrations/000002_assets.up.sql [hunk 1]
```diff
@@ -315,8 +315,7 @@ CREATE TABLE IF NOT EXISTS asset_seedlings (
 CREATE VIEW genesis_info_view AS
     SELECT
         gen_asset_id, asset_id, asset_tag, assets_meta.meta_data_hash meta_hash,
-        output_index, asset_type, genesis_points.prev_out prev_out, 
-        chain_txns.txid anchor_txid, block_height
+        output_index, asset_type, genesis_points.prev_out prev_out, block_height
```
[the remark this answers is not in the document]
Comment (jharveyb) [reply tapdb/sqlc/migrations/000002_assets.up.sql:318]: Agree, best to leave untouched from now on.

[the remark this answers is not in the document]
Comment (GeorgeTsagk) [reply tapdb/sqlc/migrations/000002_assets.up.sql:318]: Ok, I thought I had to restore this to its original state. Fixing

**Comment by GeorgeTsagk** [comment]
The above seems to fail the `postgres` unit tests with
```
ERROR: cannot drop view genesis_info_view because other objects depend on it (SQLSTATE 2BP01)
```

Is the approach correct? Seems like dropping the view before re-creating it invalidates other stuff

**Comment by jharveyb** [comment]
From offline discussion:

The `key_group_info_view` depends on this view and therefore also needs to be dropped and re-created in this migration.

**Comment by jharveyb** [comment]
Ok, I reproduced the original error from #685 with approx. these steps:

- Spin up two tapds at version v0.3.0 with separate backing lnds.
- Mint an asset on tapd 1 and tapd 2.
- Add tapd 1 as a federation member for tapd 2.
- Restart tapd 2 at version v0.3.1.
- Make an address on tapd 2 for the asset minted on tapd 1. This should succeed.
- Restart tapd 2.
- Trying to create any address on tapd 2 should now fail.

Restarting tapd 2 after building this branch worked and resolved the issue, but from manual inspection the DB seems to be in a weird state. The `schema migrations` table shows version 11 not 12 for example, and I don't see the new row in the `genesis_info_view`.

Not sure if `000012_anchor_txid_down.sql` should be non-empty since we have the `DROP` commands in the up migration.

**Comment by GeorgeTsagk** [comment]
> Not sure if 000012_anchor_txid_down.sql should be non-empty since we have the DROP commands in the up migration.

Isn't the `down` migration supposed to revert the effects of the `up`? If running `up` leaves you with 2 views then why would `down` not try to delete those?

**Comment by jharveyb** [comment]
Double-checked, I think the only migration missing a `down`  is 000010, which also recreates a view. I'm not sure exactly why not having a `down` migration there works.

File: tapdb/sqlc/migrations/000002_assets.up.sql [hunk 2]
```diff
@@ -315,7 +315,7 @@ CREATE TABLE IF NOT EXISTS asset_seedlings (
 CREATE VIEW genesis_info_view AS
     SELECT
         gen_asset_id, asset_id, asset_tag, assets_meta.meta_data_hash meta_hash,
-        output_index, asset_type, genesis_points.prev_out prev_out, 
+        output_index, asset_type, genesis_points.prev_out prev_out,
```
Comment (ffranr) [inline tapdb/sqlc/migrations/000002_assets.up.sql:318]: I don't think we should modify an old migration file. Might make it more complicated to debug later.

**Review by ffranr — COMMENTED** [review]
I don't think we should worry about rolling back migrations. I think it's ok to leave the `.down.sql` file empty.

Is it possible to add an itest that would break without this new view? I can't see why our itests didn't catch this to begin with.

File: tapdb/sqlc/migrations/000002_assets.up.sql (hunk 2 above)
Comment (GeorgeTsagk) [reply tapdb/sqlc/migrations/000002_assets.up.sql:318]: This PR only has a trailing space as diff. Are you suggesting we revert the contents of this migration to the ones present prior to #667 ? (remove `anchor_txid` from **this** PR?)

**Review by ffranr — APPROVED** [review]
Looks good to me! :+1:

**Review by guggero — COMMENTED** [review]
Post merge ACK.

**Pull request merged**